services/src/datasets/external/nature40.rs

```diff
@@ -284,6 +289,31 @@ impl Nature40DataProvider {
         }
         Ok(txt)
     }
+
+    fn parse_time_positions(dataset: &gdal::Dataset) -> Result<Vec<String>> {
+        let mut reader = Reader::from_file(&dataset.description()?)?;
```

Comment:
can you please try to avoid reading the dataset description file twice? You should be able to load it once and use it for getting the labels *and* the time steps